added companies and their works of different technology in motivation…

yati1998 opened the pull request:
Fixes issue #852

alodahl commented on algorithms/intro-to-algorithms.md:
```diff
@@ -29,6 +29,15 @@ Recommended:
+ **Which companies use media queries?**
+Everyone. Some companies with very responsive sites are:
+- [Siri](https://www.apple.com/siri/) 
+  -  How does it work, you ask? After you give Siri a command or ask it a question, the algorithm puts your voice into text to code it. Siri then dissects the code to pick up on specific keywords and phrases to figure out the context of the command. After the context is decided, the algorithm then decides if the answer can be found within the account’s personal data (contacts, saved photos, music ...
```
If you are going to use someone else’s words, you need to cite the source.  http://www.tilr.com/blog/5-algorithm-technologies-you-use-everyday/

alodahl commented on algorithms/intro-to-algorithms.md:
```diff
@@ -29,6 +29,15 @@ Recommended:
+- [Siri](https://www.apple.com/siri/) 
+  -  How does it work, you ask? After you give Siri a command or ask it a question, the algorithm puts your voice into text to code it. Siri then dissects the code to pick up on specific keywords and phrases to figure out the context of the command. After the context is decided, the algorithm then decides if the answer can be found within the account’s personal data (contacts, saved photos, music ...
+- [Netflix](https://www.netflix.com/nl-en/).
+  - The video streaming giants over at Netflix have developed a personalization algorithm that perfectly blends what you want to watch and new suggestions that will turn into one of your favorites. Netflix has also developed a unique algorithm that tackles any streaming service’s most frustrating problem: buffering. Their algorithm is able to determine each video frame and compact it to lowest po ...
```
http://www.tilr.com/blog/5-algorithm-technologies-you-use-everyday/

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,13 @@
 
+### Motivation
+
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report.
```
https://www.quora.com/Why-is-sorting-important-in-computer-science-and-programming

alodahl commented on navigating-new-codebases/navigating-new-codebases.md:
```diff
@@ -20,7 +20,7 @@ PHP (Larvel)
 Scala (Play)
 
-Companies such as GitHub and Airbnb uses Ruby on Rails whereas Instagram, Dropbox, and Spotify use Django. Most, if not all, of the source code has already been written - as a web developer, you will need to quickly interpret the code and either add to the code or debug any issues associated with the code.
+Companies such as [GitHub](https://www.github.com) and [Airbnb](https://www.airbnb.co.in/) uses Ruby on Rails whereas Instagram, Dropbox, and Spotify use Django. Most, if not all, of the source code has already been written - as a web developer, you will need to quickly interpret the code and either add to the code or debug any issues associated with the code.
```
👏🏻

alodahl commented on open-source/open-source.md:
```diff
@@ -15,6 +15,14 @@ Contributing to Open Source projects offers many benefits:
 
+**Which companies uses open source?**
+- [Adobe](https://www.adobe.io/)
+  - Adobe has a strong commitment to open source and has more than 250 public repositories on its GitHub site. Some of its best-known open source projects are developer tools like the PhoneGap web development framework, the Brackets text editor and the Topcoat CSS library. Adobe staff also contribute regularly to other open source projects like Gecko, Blink, WebKit, Apache Cordova, Flex, Felix an ...
```
https://www.datamation.com/amp/open-source/35-top-open-source-companies-1.html

alodahl commented on open-source/open-source.md:
```diff
@@ -15,6 +15,14 @@ Contributing to Open Source projects offers many benefits:
+- [Adobe](https://www.adobe.io/)
+  - Adobe has a strong commitment to open source and has more than 250 public repositories on its GitHub site. Some of its best-known open source projects are developer tools like the PhoneGap web development framework, the Brackets text editor and the Topcoat CSS library. Adobe staff also contribute regularly to other open source projects like Gecko, Blink, WebKit, Apache Cordova, Flex, Felix an ...
+- [Automattic](https://automattic.com/)
+  -  You might not have heard of Automattic before, but you have almost certainly heard of its best-known creation — WordPress. According to the website, WordPress currently "powers 28 percent of the Internet." Although the WordPress open source project is now handled by a foundation, Automattic continues to contribute to its code, and it runs WordPress.com. It also participates in many other ope ...
```
https://www.datamation.com/amp/open-source/35-top-open-source-companies-1.html

yati1998 commented:
Shall I remove the description and just add the links.?

yati1998 commented:
@alodahl  shall I remove the description and only add the links instead?

yati1998 commented:
@alodahl I have made a new commit. Please do check that.

alodahl commented:
You can keep the text, Just make sure it’s clear that the text is from that site. Always do this any time you use someone else’s text or images, or it’s plagiarism. If you’ve done it anywhere else, please change it now.

yati1998 commented:
No I haven't done that else where.  @alodahl I have already made the change and removed the text and added link instead.  Will that be fine?  Or shall I change it back?  Please do review my PR.  So that I can start working on the other issue

alodahl commented:
> No I haven't done that else where. @alodahl I have already made the change and removed the text and added link instead. Will that be fine? Or shall I change it back? Please do review my PR. So that I can start working on the other issue

 https://github.com/Techtonica/curriculum/pull/842#pullrequestreview-240683006

alodahl commented on algorithms/intro-to-algorithms.md:
```diff
@@ -29,6 +29,12 @@ Recommended:
 
 Barring that, you'll need to know algorithms because designing the wrong one will probably mean your code will be slow crazy. They're also commonly referenced in engineering job interviews, and if you don't pass that then you won't get the job. If you're comfortable with both of these, you can skip this course.
+ 
+ **Which companies use media queries?**
```
Media queries in the algorithms outline?

yati1998 replied:
> Media queries in the algorithms outline?

Oh, Sorry Haven't noticed that. I will change it.

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,15 @@
+
+For example bubble sort, quick sort, and selection sort all take the same input and produce the same output but at different speeds. This is a nice way to show how different approaches can yield more efficient algorithms using approaches such as divide and conquer. On top of that, sorting algorithms don't require advanced data structures knowledge to understand. If you can understand arrays you c ...
+
+- [Why is sorting important in computer science?](https://www.quora.com/Why-is-sorting-important-in-computer-science-and-programming)
```
Change this to something like: 
- [from: Why is sorting important in computer science? by *Someone*]()

yati1998 replied:
@alodahl sorry but i couldnt get this. Can you please explain?

alodahl replied:
Because you need to make it clear that this whole block of text is from that site

yati1998 commented:
@alodahl I have added the credits. Do mention further changes.

saberprashant commented on algorithms/intro-to-algorithms.md:
```diff
@@ -29,6 +29,12 @@ Recommended:
 Barring that, you'll need to know algorithms because designing the wrong one will probably mean your code will be slow crazy. They're also commonly referenced in engineering job interviews, and if you don't pass that then you won't get the job. If you're comfortable with both of these, you can skip this course.
+ 
+ **Which companies use algorithms?**
+Everyone. Some companies with very responsive sites are:
```
How is, "responsive sites" helpful over here?

saberprashant commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,15 @@
+### Motivation
+
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report.
+The point isn't to remember the sorting algorithms so you can write them on the job. We of course have library algorithms/search engines for that as you said. They are taught because they serve as a good case studies on algorithm analysis/design. Of course this doesn't mean its the best way (but theres already a lot of material out there on it for teaching purposes so it's probably easier for pro ...
```
@yati1998 , please rephrase this para. Make it easier and concise to understand.
For example, start like this:
The point here isn't to remember the sorting algorithms which you can write on your job or in an interview, but, to understand what an algorithm actually is.

saberprashant replied:
@yati1998 please rephrase this para, try to make it easier and concise to understand.
For example, start like this:
The point here isn't to remember the sorting algorithms which you can write on your job or in an interview, but, to understand what an algorithm actually is.

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,15 @@
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report.
+The point here isn't to remember the sorting algorithms which you can write on your job or in an interview, but, to understand what an algorithm actually is. We already have different libraries for this purpose but still, these algorithms serve as a good case study on algorithm analysis/design.
+
+For example bubble sort, quick sort, and selection sort all take the same input and produce the same output but at different speeds. This is a nice way to show how different approaches can yield more efficient algorithms using approaches such as divide and conquer. On top of that, sorting algorithms don't require advanced data structures knowledge to understand. If you can understand arrays you c ...
```
https://dev.to/cooperfle/whats-the-point-of-learning-how-to-sort-20dl   ☹️

yati1998 replied:
I have written this from https://www.quora.com/Why-is-sorting-important-in-computer-science-and-programming

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,15 @@
+### Motivation
+
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report.
+The point here isn't to remember the sorting algorithms which you can write on your job or in an interview, but, to understand what an algorithm actually is. We already have different libraries for this purpose but still, these algorithms serve as a good case study on algorithm analysis/design.
```
can you remove this sentence?  I don't think it adds to the section.  "We already have different libraries for this purpose but still, these algorithms serve as a good case study on algorithm analysis/design."

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,15 @@
+
+For example bubble sort, quick sort, and selection sort all take the same input and produce the same output but at different speeds. This is a nice way to show how different approaches can yield more efficient algorithms using approaches such as divide and conquer. On top of that, sorting algorithms don't require advanced data structures knowledge to understand. If you can understand arrays you c ...
+
+- [Why is sorting important in computer science? By:Siddharth Khuntwal](https://www.quora.com/Why-is-sorting-important-in-computer-science-and-programming)
```
this is incorrect.  put it with the text you quoted, or note that they should look below for the source somehow - no one can tell what this is citing.

alodahl commented on algorithms/intro-to-algorithms.md:
```diff
@@ -29,6 +29,12 @@ Recommended:
 Barring that, you'll need to know algorithms because designing the wrong one will probably mean your code will be slow crazy. They're also commonly referenced in engineering job interviews, and if you don't pass that then you won't get the job. If you're comfortable with both of these, you can skip this course.
+ 
+ **Which companies use algorithms?**
+Everyone. Some of the companies are:
```
Everyone? Really? Plus, this is a sentence I wrote. Please rephrase this.

yati1998 replied:
@alodahl I have made the changes. Please do review them.

alodahl commented on algorithms/sorting.md:
```diff
@@ -8,6 +8,12 @@
 
+### Motivation
+
+Sorting is important in programming for the same reason it is important in everyday life. It is easier and faster to locate items in a sorted list than unsorted. Sorting algorithms can be used in a program to sort an array for later searching or writing out to an ordered file or report. [BY:Siddharth Khuntwal](https://www.quora.com/Why-is-sorting-important-in-computer-science-and-programming)
```
thank you!
by Siddarth Khuntwal - please fix formatting

alodahl commented on interviewing/technical-interviewing.md:
```diff
@@ -25,6 +25,11 @@ on solving technical questions posed by an interviewer instead of
 trying to recall a basic concept of programming, the more confident you
 will feel and the more successful you'll be.
 
+**Which companies focuses on technical and coding skills?**
```
focus

alodahl commented on interviewing/technical-interviewing.md:
```diff
@@ -25,6 +25,11 @@ on solving technical questions posed by an interviewer instead of
 will feel and the more successful you'll be.
 
+**Which companies focuses on technical and coding skills?**
+- [Amazon](https://www.makeuseof.com/tag/si-coding-challenges-lead-money-jobs/)
```
Can you provide a sentence on what they will find at each resource?

yati1998 replied:
@alodahl each of the resources says how technology is used by the companies. So, can I directly mention before the companies: "the below links describe the use of technology in the company"

alodahl replied:
I think this is a weird way to list it, since there is no description about these specific companies in the link.  But there is a description of the hackerrank challenge.  Can you change this to describing how the challenge in the link could lead to jobs at Amazon, Redhat, and Uber?

alodahl commented on open-source/open-source.md:
```diff
@@ -15,6 +15,11 @@ Contributing to Open Source projects offers many benefits:
 - Offers a new perspective on working on code outside of your usual place (like through work)
 
+**Which companies uses open source?**
+- [Adobe](https://www.datamation.com/amp/open-source/35-top-open-source-companies-1.html)
```
describe what they will find at each link.

alodahl commented on roles-in-tech/data-science.md:
```diff
@@ -16,7 +16,7 @@ No Prerequisites.
     * How can you best predict future performance?
 
-* **Netflix** uses data analysis to recommend movies.
+* **[Netflix](https://www.netflix.com/)** uses data analysis to recommend movies.
```
👍🏼

alodahl commented on roles-in-tech/developer-advocate.md:
```diff
@@ -10,9 +10,9 @@
 
 ### Motivation
-A developer evangelist is a spokesperson, mediator and translator between a company and both its technical staff and outside developers. It's not always about writing code. What if you don't write code?
+A developer evangelist is a mediator or translator between a company and the developers. Even if you don't write codes, you can get into your dream companies.
```
code
company

alodahl commented on roles-in-tech/security.md:
```diff
@@ -18,6 +18,12 @@
 
+**Comapnies that use Security**
+- [Blackberry](http://www.blackberry.com/)
+  - It's been a long time since BlackBerry was known as the handheld device icon. These days it's doing a brisk business in the cybersecurity market, recently inking a $1.4  billion acquisition of Cylance to further its goal to become the leader in providing end-to-end mobility services that are secure and trusted.
```
cite the exact url where you found each quote, after the quote.

yati1998 commented:
@alodahl I have made the required changes. Hope it works this time.

yati1998 commented:
@alodahl I guess you are seeing the previous commit. I have added a line to describe the link and also changed it to focus. Still, I am adding the hackerrank stuff.

alodahl commented on interviewing/technical-interviewing.md:
```diff
@@ -22,10 +22,9 @@ Regardless of a person's experience level, it's important to be relatively comfo
-- [Amazon](https://www.makeuseof.com/tag/si-coding-challenges-lead-money-jobs/)
-- [RedHat](https://www.makeuseof.com/tag/si-coding-challenges-lead-money-jobs/)
-- [Uber](https://www.makeuseof.com/tag/si-coding-challenges-lead-money-jobs/)
+HackerRank is one of those websites or platforms that will help you get to into you dream companies. It conducts various coding competitions and hackathons through which coders directly get into those companies. It also has practice questions to help you improve your coding skills. It conduct contests on algorithms, machine learning and many new trending technologies.
```
Great!  Don't forget the link.

yati1998 commented:
@alodahl Please review my changes. And I am sorry for the error in the commit message, I just noticed now. I will take care of it next time.

alodahl commented on navigating-new-codebases/navigating-new-codebases.md:
```diff
@@ -18,6 +18,7 @@ N/A
 
 - Companies such as GitHub and Airbnb uses Ruby on Rails whereas Instagram, Dropbox, and Spotify use Django. Most, if not all, of the source code, has already been written - as a web developer, you will need to quickly interpret the code and either add to the code or debug any issues associated with the code.
 
+
```
Mentors: no points for this white space

yati1998 replied:
@alodahl sorry but couldn't get it. Do you want me to remove this extra blank space?

alodahl replied:
Thank you for doing that, but it’s not part of an issue, that’s all.

alodahl commented on roles-in-tech/security.md:
```diff
@@ -18,6 +18,12 @@
 
 Apprentices will learn secure development basics, common pitfalls, and how to avoid them.
 
+**Comapnies that use Security**
```
Companies

yati1998 commented:
@alodahl  am I done with this now?

alodahl commented:
@saberprashant are you ok with this?

alodahl commented:
Yati, please resolve the conflicts so we can merge.
